system/Database/BaseConnection.php

```diff
@@ -1238,6 +1238,10 @@ public function escape($str)
         }
 
         if (is_string($str) || (is_object($str) && method_exists($str, '__toString'))) {
```

Comment:
Do we have other database related classes which have the `__toString()` that can be passed here as query string.?

Reply:
I'm not sure, I just thought it was there because such objects would be compatible.

Reply:
From the looks of it you could create a wrapper class for DateTime with __toString() and use it here. I thought it would be nice to build some sort of class that could work with each DB's date format.. and then also have a frontend format.